cmd/algorand-indexer/daemon.go

```diff
@@ -120,123 +103,20 @@ var daemonCmd = &cobra.Command{
-	ptr   *bool
-}
-
-func (sv configBoolVar) Set(x string) {
-	// only set if it still has the original value
-	if *sv.ptr != sv.value {
-		return
-	}
-	xl := strings.ToLower(x)
-	for _, ts := range trueStrings {
-		if ts == xl {
-			*sv.ptr = true
-			return
-		}
-	}
-	*sv.ptr = false
-}
-
-func configBoolVarP(flags *pflag.FlagSet, boolPtr *bool, name, short string, value bool, usage string) {
-	*boolPtr = value
-	flags.BoolVarP(boolPtr, name, short, value, usage)
-	configVars = append(configVars, configVar{name, short, usage, &configBoolVar{value, boolPtr}})
-}
-
-// TODO: maybe someday replace file parsing with YAML library, but for now we don't need nested structure and a smaller dependency tree makes me happy
-func configFromStream(in io.Reader) (err error) {
-	lineno := 0
-	lineReader := bufio.NewReader(in)
-	for true {
-		linebytes, isPrefix, err := lineReader.ReadLine()
-		if err == io.EOF {
-			return nil
-		}
-		lineno++
-		if err != nil {
-			return err
-		}
-		if isPrefix {
-			return fmt.Errorf(":%d line too long", lineno)
-		}
-		if len(linebytes) == 0 {
-			continue
-		}
-		if linebytes[0] == '#' {
-			continue
-		}
-		line := string(linebytes)
-		colon := strings.IndexRune(line, ':')
-		if colon < 0 {
-			return fmt.Errorf(":%d line is not \"key: value\"", lineno)
-		}
-		key := strings.TrimSpace(line[:colon])
-		value := strings.TrimSpace(line[colon+1:])
-		ok := false
-		for _, cvar := range configVars {
-			if cvar.name == key {
-				cvar.t.Set(value)
-				ok = true
-				break
-			}
-		}
-		if !ok {
-			return fmt.Errorf(":%d unknown key %s", lineno, key)
-		}
-	}
-	return nil
-}
-
 func init() {
-	configStringVarP(daemonCmd.Flags(), &algodDataDir, "algod", "d", "", "path to algod data dir, or $ALGORAND_DATA")
-	configStringVarP(daemonCmd.Flags(), &algodAddr, "algod-net", "", "", "host:port of algod")
-	configStringVarP(daemonCmd.Flags(), &algodToken, "algod-token", "", "", "api access token for algod")
-	configStringVarP(daemonCmd.Flags(), &genesisJsonPath, "genesis", "g", "", "path to genesis.json (defaults to genesis.json in algod data dir if that was set)")
-	configStringVarP(daemonCmd.Flags(), &daemonServerAddr, "server", "S", ":8980", "host:port to serve API on (default :8980)")
-	configBoolVarP(daemonCmd.Flags(), &noAlgod, "no-algod", "", false, "disable connecting to algod for block following")
-	configStringVarP(daemonCmd.Flags(), &tokenString, "token", "t", "", "an optional auth token, when set REST calls must use this token in a bearer format, or in a 'X-Indexer-API-Token' header")
-	configBoolVarP(daemonCmd.Flags(), &developerMode, "dev-mode", "", false, "allow performance intensive operations like searching for accounts at a particular round")
-
-	daemonCmd.Flags().StringVarP(&configFilePath, "config", "c", "", "path to 'key: value' config file, keys are same as command line options")
+	logger = log.New()
+	logger.SetFormatter(&log.JSONFormatter{})
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(log.InfoLevel)
```

Comment:
Should probably be configurable (or maybe dev mode should enable a lower level one)

Reply:
-v/--verbose is a lovely convention

Reply:
This isn't really new, there were 2 init functions that I merged together.

I'm going to improve the logger situation in #179